src/ol/style/IconImage.js

```diff
@@ -152,18 +159,22 @@ class IconImage extends EventTarget {
    * @return {HTMLImageElement|HTMLCanvasElement} Image element.
    */
   getHitDetectionImage(pixelRatio) {
-    if (!this.hitDetectionImage_) {
-      if (this.isTainted_()) {
+    if (!this.hitDetectionImage_[pixelRatio]) {
+      if (this.color_) {
```

Comment:
The isTainted check should not be removed. If the image is loaded from an external server or is and svg in internet explorer the canvas data will not be accessible.

Reply:
Then I also modify the replaceColor function to have all the canvas function in the isTainted if, Thanks :-)